### DISCUSSION TITLE
I am having difficulty in installation

### DISCUSSION THREAD

**Discussion by DimitrisElGreco** [seemoo-lab/nexmon#549] [Q&A]
Can anyone please provide a guide for getting a phone to monitor mode and packet injection.
I am trying to install nexmon firmware on a samsung galaxy s4 (whch has bcm4335b0 chip).

**Comment by psxpetey** [comment]
The way I understand it is Build patches for bcm4330, bcm4339 and bcm4358 using a x86 computer running Linux (e.g. Ubuntu 16.04)
Install some dependencies: sudo apt-get install git gawk qpdf adb flex bison

Only necessary for x86_64 systems, install i386 libs:

sudo dpkg --add-architecture i386
sudo apt-get update
sudo apt-get install libc6:i386 libncurses5:i386 libstdc++6:i386
Clone our repository: git clone https://github.com/seemoo-lab/nexmon.git

In the root directory of the repository: cd nexmon

Setup the build environment: source setup_env.sh
Compile some build tools and extract the ucode and flashpatches from the original firmware files: make
Go to the patches folder of your target device (e.g. bcm4339 for the Nexus 5): cd patches/bcm4339/6_37_34_43/nexmon/ if you have 4335b0 cd to patches/bcm4335b0/nexmon

Compile a patched firmware: make < than you do the command make in that directory
Generate a backup of your original firmware file: make backup-firmware < you run this command with you phone plugged in maybe?
Install the patched firmware on your smartphone: make install-firmware (make sure your smartphone is connected to your machine beforehand) now you run make install-firmware and its supposed to go to your phone

**Reply by psxpetey** [reply]
im about to try this tomorrow since my z3 is 4339
